operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,243 @@
+package v1alpha1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Kueue is the CRD to represent the kueue operator
+// This CRD defines the configuration that the Kueue
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kueue,scope=Cluster
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// kueue configuration must not be changed once the object exists
+	// to change the configuration, one can delete the object and create a new object.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="values are immutable once set"
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config is the desired configuration
+	// for the kueue operator.
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+// +kubebuilder:validation:Enum=QueueNameOptional;QueueNameRequired
+type ManageJobsWithoutQueueNameOption string
+
+// ManageJobsWithoutQueueName allows to control what kind of
+// jobs kueue will manage.
+// Kueue jobs usually require kueue.x-k8s.io/queue-name on each job
+// to be opt-in for Kueue.
+const (
+	// QueueNameOptional means kueue will assume all workloads
+	// are to be gated by Kueue.
+	// This must be used with ManagedJobsNamespaceSelector.
+	QueueNameOptional ManageJobsWithoutQueueNameOption = "QueueNameOptional"
+	// QueueNameRequired means that the jobs require a queue label.
+	QueueNameRequired ManageJobsWithoutQueueNameOption = "QueueNameRequired"
+)
+
+// +kubebuilder:validation:Enum=Enabled;Disabled
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// integrations are the types of integrations Kueue will manage
```

Comment:
What are integrations? Why would I, as a user, care to configure this?

Generally the GoDoc for fields should be end-user friendly. Some useful guidelines for writing GoDoc for APIs: https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc